crates/ruff_cli/src/commands/check.rs

```diff
@@ -192,9 +196,19 @@ fn lint_path(
     noqa: flags::Noqa,
     fix_mode: flags::FixMode,
     unsafe_fixes: UnsafeFixes,
+    extension_override: &FxHashMap<String, Language>,
```

Comment:
I think this should probably be on `settings`, rather than a standalone argument. That would also allow users to set this in their settings file (`pyproject.toml`, etc.), which seems important.

Reply:
Do we want to allow that as it's currently a hidden CLI argument?

Reply:
I went back and forth on this myself (see commit history!) but I thought if it was hidden it didn't make sense to be a setting, or possibly I couldn't see any examples of hidden settings so I didn't know how to make one.

Reply:
I think the `--line-length` is an example of a hidden CLI flag but can be configured via the [config file](https://docs.astral.sh/ruff/settings/#line-length):

https://github.com/astral-sh/ruff/blob/2d5ce4532ade11710ff854516cd085b9ea629cf2/crates/ruff_cli/src/args.rs#L274-L276

Reply:
Ah ok. `line_length` is still documented in `ruff_workspace`, so it's hidden in the CLI but visible in other documentation and in the schema.  Is that OK for this option?

Reply:
I moved it to settings